src/addrman.cpp

```diff
@@ -501,8 +504,14 @@ void CAddrMan::GetAddr_(std::vector<CAddress>& vAddr, size_t max_addresses, size
         assert(mapInfo.count(vRandom[n]) == 1);
 
         const CAddrInfo& ai = mapInfo[vRandom[n]];
-        if (!ai.IsTerrible())
-            vAddr.push_back(ai);
+
+        // Filter by network (optional)
+        if (network != std::nullopt && ai.GetNetClass() != network) continue;
```

Comment:
Won't this result in infnite loops, if there are not enough of the requested net class?

I think doing this correctly requires keeping track of the counts per network.

Reply:
If the early exit isn't triggered (e.g. `getnodeaddresses 10000 i2p`) the loop terminates after it has iterated through vRandom.

Reply:
Today's computers are so fast that even the infinite loop finishes in about 6 seconds.

Reply:
(I think the added functional tests here cover it.)